pkg/cloudprovider/types.go

```diff
@@ -41,6 +42,16 @@ var (
 
 type DriftReason string
 
+type RepairStatement struct {
+	// Type of unhealthy state that is found on the node
+	Type corev1.NodeConditionType
+	// Status condition when a node is unhealthy
+	Status corev1.ConditionStatus
+	// TolerationDuration is the duration the controller will wait
+	// before force terminating nodes that are unhealthy.
+	TolerationDuration time.Duration
```

Comment:
I wonder if we should consider doing tolerationSeconds, since it's a bit easier to serialize and follows the pod tolerations semantic, but I wonder how long these tolerations are realistically going to be. Thoughts on one or the other? 

I also question if toleration is really the best word we can do here. It's nice because it implies that it works the same way as taints and tolerations. Could we name it `RepairAfter`? It would match with our other `__After` semantics.

Reply:
> I wonder if we should consider doing tolerationSeconds, since it's a bit easier to serialize and follows the pod tolerations semantic, but I wonder how long these tolerations are realistically going to be. Thoughts on one or the other?

These values are expected by in ball park of 30 min. So it would feel awkward to name the field `tolerationSeconds`. This was something we went back and forth on in the RFC. I'm sure it really matter much since this is not a customer exposed field, since I'm assuming you don't mean `consolidateAfter`. If other feel a strong way I can change it.

Reply:
yeah like consolidateAfter or expireAfter. Personally I think it follows Karpenter convention to do repairAfter